src/command_parser.c

```diff
@@ -183,41 +183,51 @@ static int _ScriptRunCommand_ParseArgs(RedisModuleCtx *ctx, RedisModuleString **
     *runkey = argv[argpos];
 
     const char *arg_string = RedisModule_StringPtrLen(argv[++argpos], NULL);
-    if (!strcasecmp(arg_string, "TIMEOUT") || !strcasecmp(arg_string, "INPUTS")) {
+    if (!strcasecmp(arg_string, "TIMEOUT") || !strcasecmp(arg_string, "INPUTS") ||
+        !strcasecmp(arg_string, "OUTPUTS")) {
         RAI_SetError(error, RAI_ESCRIPTRUN, "ERR function name not specified");
         return REDISMODULE_ERR;
     }
     *func_name = arg_string;
-    arg_string = RedisModule_StringPtrLen(argv[++argpos], NULL);
 
-    // Parse timeout arg if given and store it in timeout
-    if (!strcasecmp(arg_string, "TIMEOUT")) {
-        if (_parseTimeout(argv[++argpos], error, timeout) == REDISMODULE_ERR)
-            return REDISMODULE_ERR;
-        arg_string = RedisModule_StringPtrLen(argv[++argpos], NULL);
-    }
-    if (strcasecmp(arg_string, "INPUTS") != 0) {
-        RAI_SetError(error, RAI_ESCRIPTRUN, "ERR INPUTS not specified");
-        return REDISMODULE_ERR;
-    }
-
-    bool is_input = true, is_output = false;
+    bool is_input = false;
+    bool is_output = false;
+    bool timeout_set = false;
     size_t ninputs = 0, noutputs = 0;
     int varidic_start_pos = -1;
 
     while (++argpos < argc) {
         arg_string = RedisModule_StringPtrLen(argv[argpos], NULL);
+
+        // Parse timeout arg if given and store it in timeout
+        if (!strcasecmp(arg_string, "TIMEOUT") && !timeout_set) {
+            if (_parseTimeout(argv[++argpos], error, timeout) == REDISMODULE_ERR)
+                return REDISMODULE_ERR;
+            timeout_set = true;
+            continue;
+        }
+
+        if (!strcasecmp(arg_string, "INPUTS") && !is_input) {
+            is_input = true;
```

Comment:
I think that this way, INPUTS can appear after OUTPUTS (even more than once...)
Consider add another flag like "input_done"?